src/Microsoft.DotNet.Github.IssueLabeler/Models/Labeler.cs

```diff
@@ -20,36 +20,21 @@ public class Labeler
         private readonly string _repoOwner;
         private readonly string _repoName;
         private readonly double _threshold;
-        private readonly string _clientId;
-        private readonly string _clientSecret;
         private readonly string _secretUri;
 
-        public Labeler(string repoOwner, string repoName, string clientId, string clientSecret, string secretUri, double threshold)
+        public Labeler(string repoOwner, string repoName, string secretUri, double threshold)
         {
             _repoOwner = repoOwner;
             _repoName = repoName;
             _threshold = threshold;
-            _clientId = clientId;
-            _clientSecret = clientSecret;
             _secretUri = secretUri;
         }
 
         private async Task GitSetupAsync()
         {
-            var kvc = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(
-                async (authority, resource, scope) =>
-                {
-                    var authContext = new AuthenticationContext(authority);
-                    var clientCred = new ClientCredential(_clientId, _clientSecret);
-                    var result = await authContext.AcquireTokenAsync(resource, clientCred);
-
-                    if (result == null)
-                        throw new InvalidOperationException("Failed to obtain the github token");
-
-                    return result.AccessToken;
-                }));
-
-            SecretBundle secretBundle = await kvc.GetSecretAsync(_secretUri);
+            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
```

Comment:
Unless there's an ambiguity with an overload, this can be simplified to

```suggestion
            KeyVaultClient keyVaultClient = new KeyVaultClient(azureServiceTokenProvider.KeyVaultTokenCallback);
```

because of delegate inference (or whatever the feature is called), from C# 2.0.

Reply:
The change results in a type mismatch compile error

Reply:
Fair enough.  Either there's a co/contra-variance alignment that the `new DelegateType(methodGroup)` syntax is forcing, or they did bad overloading and there's an ambiguous target from that methodgroup input.

Either way, if this is required, it's required :smile:.